### ISSUE TITLE
Wi-Fi disconnects during transfer from GoPro

### ISSUE THREAD

**Issue by talllguy** [mapillary/mapillary_issues#1740]
| Platform | Version |
| ------------- | ------------- |
| Smartphone | iPhone 6 Plus |
| OS | iOS 9.2.1 |
| Free space | 25 GB |
| Mapillary app  | 4.1.5 *beta* |
| Camera | GoPro Hero 4 Silver |

I have observed a strange behavior with the latest beta build. After transferring around 50 photos, the phone disconnects from the GoPro Wi-Fi. Even weirder is that the number of photos transfered progress bar continues to increment, while the Wi-Fi is disconnected.

I first observed this when I noticed that my phone was on my home's Wi-Fi, as indicated by the Wi-Fi connection indicator, yet the transfer progress was still going. :thought_balloon: After one time when the transfer failed, the app asked for my Hero 3 Password, but I don't have a Hero 3. So I wonder if maybe that new Hero 3 code has something to do with this? 

Steps to troubleshoot so far:

1. I canceled the transfer
2. Reconnected to GoPro Wi-Fi
3. Resumed transferring.
4. Wait a few minutes
5. Observe that the regular home Wi-Fi is back.

Next I "forgot" my home Wi-Fi.. :exclamation: what the heck is my password :question: so as to prevent the phone from going back to it.

1. Connect to GoPro Wi-Fi
2. Resume transfer
3. Wait an hour for all ~3000 photos to transfer
4. Done!
5. *Wait a minute here*
6. Only ~250 or so photos were transferred.

One more troubleshooting attempt:

1. Full restart iPhone
2. Repeat above
3. Write this up on Github :octocat: 

Here are a few screenshots that show this strange behavior. *Note: descriptions with timestamps are above images.*

## 2016-03-09 20 11 05: Transfer looks normal
![2016-03-09 20 11 05](https://cloud.githubusercontent.com/assets/1827881/13658829/acae2cd0-e648-11e5-9d7f-0b9ad79d45ca.png)

## 2016-03-09 20 11 16: But wait... Look Ma, no Wi-Fi! :older_woman: 
![2016-03-09 20 11 16](https://cloud.githubusercontent.com/assets/1827881/13658833/b8afbc42-e648-11e5-8ffd-900b4d79dfd7.png)

## 2016-03-09 20 11 30: No networks connected
![2016-03-09 20 11 30](https://cloud.githubusercontent.com/assets/1827881/13658841/d17dd0ba-e648-11e5-91e1-36809124f9ff.png)

## 2016-03-09 20 11 43: App still thinks it is transferring
![2016-03-09 20 11 43](https://cloud.githubusercontent.com/assets/1827881/13658854/edd0cdbc-e648-11e5-8eb7-ef7bb2708943.png)

## 2016-03-09 20 11 49: After cancelling, I get this message
![2016-03-09 20 11 49](https://cloud.githubusercontent.com/assets/1827881/13658863/06ec0442-e649-11e5-8582-bac45f2c10a2.png)

cc: @millenbop

**Comment by millenbop** [comment]
Thanks a bunch for reporting! And thanks for the awesome bug report.

Did you connect the GoPro to a computer? Have you turned off sleep on the GoPro?

**Comment by talllguy** [comment]
@millenbop I only connected it via USB to a little charging brick. I reset the settings on the GoPro right before setting out, so sleep was disabled.

## Update

Tried again today, after restarting the camera. I also turned bluetooth on, on iPhone, and did not have the camera connected to a charger. All 2919 photos made it across! *And I got my favorite message.*

![2016-03-10 09 35 17](https://cloud.githubusercontent.com/assets/1827881/13672807/4c0e5bbe-e6a4-11e5-9b5b-6b26b810a064.png)

**Comment by millenbop** [comment]
Hehe great! Well let me know if you observe it again!

**Comment by talllguy** [comment]
I haven't seen this one in a while. I'll close for now. May have been a one off thing.

**Comment by Virtakuono** [comment]
I am facing ton of similar issues, with Session4. I hope this is a right place to report, if not, I can open a new ticket too. I doubt that I can give as detailed a report as @talllguy tho...

I am losing connection sporadically pretty much all the time when transferring photos from my gopro to iPhone 5.I find it particularly annoying that the indicator says something like 1000/2000 photos transferred when it fails, but when I need to start from scratch again...

**Comment by talllguy** [comment]
@Virtakuono I've noticed in some cases "starting from scratch" again is faster, because the app checks the existing photos against the ones on the camera, so it doesn't take as long the second time.

**Comment by millenbop** [comment]
@Virtakuono when you say loosing connection, do you mean that it actually disconnects from the camera or just that it can't download all images?

Some other notes:
- we are reworking the syncing to be time-based instead of filename-based, which will be much more stable
- the Session4 is not officially supported, i.e. we haven't done any testing with a session at all and GoPro says they shouldn't be supported by the API or the new SDK so there is no guarantee how long it will work (but we will try to support it if it's possible anyway)

**Comment by Virtakuono** [comment]
@millenbop I don't have any photos so I could reproduce it right now, but if I remember correctly, it says that it could not download all the images. I don't remember re-establishing the connection before starting from scratch.

I guess I'll have to go shopping for a fancier GoPro next...

**Comment by talllguy** [comment]
@Virtakuono if you're seriously in the market, consider the Virb. I really prefer it in my few weeks with it so far, over the GoPro.

**Comment by millenbop** [comment]
@Virtakuono sometimes it fails to downloads images. In that case, just start the sync again and it should only try to download files that aren't downloaded.

And like I said, we will try to support the Session if it's possible.
